Sources/Turf/Turf.swift

```diff
@@ -35,9 +37,9 @@ public func intersection(_ line1: LineSegment, _ line2: LineSegment) -> Location
                                             longitude: line1.0.longitude + a * (line1.1.longitude - line1.0.longitude))
     
     /// True if line 1 is finite and line 2 is infinite.
-    let intersectsWithLine1 = a > 0 && a < 1
+    let intersectsWithLine1 = a >= 0 && a <= 1
```

Comment:
I am not sure if this was intended or not, but strict comparison here will not detect an intersection by line's ending points (for example when 2 LineStrings share the same coordinate). [turf-line-intersect](https://github.com/Turfjs/turf/blob/master/packages/turf-line-intersect/index.ts#L124) has such detection, but `Turfjs`'s `point-on-line` implementation (which seems to be used as base reference for Swift port) for some reason [does not](https://github.com/Turfjs/turf/blob/142e137ce0c758e2825a260ab32b24db0aa19439/packages/turf-point-on-line/index.js#L138). It seems like a bug to me, as when I seek for intersections between lines I usually want to include end points, especially if it can happen even with LineString's intermediate points, so I decided to fix that aswell.